pkg/utils/experimentHelper.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	litmuschaosv1alpha1 "github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateExperimentList make the list of all experiment, provided inside chaosengine
+func (engineDetails *EngineDetails) CreateExperimentList() []ExperimentDetails {
+	var ExperimentDetailsList []ExperimentDetails
+	for i := range engineDetails.Experiments {
+		ExperimentDetailsList = append(ExperimentDetailsList, engineDetails.NewExperimentDetails(i))
+	}
+	return ExperimentDetailsList
+}
+
+// NewExperimentDetails create and initialize the experimentDetail
+func (engineDetails *EngineDetails) NewExperimentDetails(i int) ExperimentDetails {
+	var experimentDetails ExperimentDetails
+	experimentDetails.Env = make(map[string]string)
+	experimentDetails.ExpLabels = make(map[string]string)
+
+	// set the initial values from the EngineDetails struct
+	experimentDetails.Name = engineDetails.Experiments[i]
+	experimentDetails.SvcAccount = engineDetails.SvcAccount
+	experimentDetails.Namespace = engineDetails.EngineNamespace
+	// Setting the JobName in Experiment related struct
+	experimentDetails.JobName = experimentDetails.Name + "-" + RandomString()
+	return experimentDetails
+}
+
+// SetValueFromChaosResources fetches required values from various Chaos Resources
+func (expDetails *ExperimentDetails) SetValueFromChaosResources(engineDetails *EngineDetails, clients ClientSets) error {
+	if err := expDetails.SetValueFromChaosEngine(engineDetails, clients); err != nil {
+		return errors.Errorf("Unable to set value from Chaos Engine, error: %v", err)
+	}
+	if err := expDetails.HandleChaosExperimentExistence(*engineDetails, clients); err != nil {
+		return errors.Errorf("Unable to get ChaosExperiment Name: %v, in namespace: %v, error: %v", expDetails.Name, expDetails.Namespace, err)
+	}
+	if err := expDetails.SetValueFromChaosExperiment(clients, engineDetails); err != nil {
```

Comment:
To make it more explicit, can we name this function as ``SetDefaultAttributeValuesFromChaosExperiment`